src/main/java/core/basesyntax/service/impl/strategy/BalanceOperationServiceImpl.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax.service.impl.strategy;
+
+import core.basesyntax.db.FruitDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationService;
+
+public class BalanceOperationServiceImpl implements OperationService<FruitTransaction> {
+    private final FruitDaoImpl fruitDao = new FruitDaoImpl();
+
+    @Override
+    public int processOperation(FruitTransaction fruitTransaction) {
+        return fruitDao.addFruits(fruitTransaction.getFruitName(),
```

Comment:
Balance operation is revision of quantity.
You should replace existing amount with new one, not add